File: backend/EduCore/EduCore.Business/Entities/Matricula.cs

namespace EduCore.Business.Entities
{
    public class Matricula
    {
        public int Id { get; private set; }
        public int AlumnoId { get; private set; }
        public Alumno Alumno { get; private set; }
        public int SeccionId { get; private set; }
        public Seccion Seccion { get; private set; }
        public DateTime FechaMatricula { get; private set; }
        public bool Activo { get; private set; }

        private Matricula() { } // Constructor para EF Core

        public Matricula(int alumnoId, int seccionId)
        {
            if (alumnoId <= 0)
                throw new ArgumentException("El alumno es obligatorio");

            if (seccionId <= 0)
                throw new ArgumentException("La sección es obligatoria");

            AlumnoId = alumnoId;
            SeccionId = seccionId;
            FechaMatricula = DateTime.Now;
            Activo = true;
        }

        public void Anular()
        {
            Activo = false;
        }
    }
}

File: frontend/educore-front/src/app/core/services/matricula.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';

export interface MatriculaDto {
  id: number;
  alumnoId: number;
  nombreAlumno: string;
  codigoAlumno: string;
  seccionId: number;
  nombreSeccion: string;
  fechaMatricula: string;
  activo: boolean;
}

export interface CrearMatriculaDto {
  alumnoId: number;
  seccionId: number;
}

@Injectable({
  providedIn: 'root'
})
export class MatriculaService {
  readonly apiUrl = `${environment.apiUrl}/Matricula`;

  constructor(private http: HttpClient) {}

  crear(dto: CrearMatriculaDto) {
    return this.http.post(this.apiUrl, dto);
  }

  anular(id: number) {
    return this.http.patch(`${this.apiUrl}/${id}/anular`, {});
  }
}